### DISCUSSION TITLE
Dec 2023: EOL Galactic still recommended over Humble?

### DISCUSSION THREAD

**Discussion by slowrunner** [iRobotEducation/create3_docs#475] [Questions] [answered]
### How are you connecting to your Create 3?

Wi-Fi (ROS 2)

### Computer(s) Model(s) and Operating System(s)

n/a

### Which version of ROS 2 is installed on your computer?

Other

### Which firmware version is installed on your robot?

n/a

### Which RMW is your robot running?

I don't know

### Does your robot have an assigned namespace? If so, please share.

n/a

### Is the robot connected to a network? If so, what is the network type?

WiFi

### Are there multiple Create 3 robots connected to your network?

None

### Is multicast enabled?

None

### What is the Adapter Board's USB/BLE Toggle currently switched to?

None

### Describe your question.

Docs say "Mar 2023 ROS 2 Galactic is recommended, more performant".  I'm not sure want more performant means.  

I really like the --once --flow-style of Humble, and since Galactic is EOL’d would like to use any ROS 2 bug fixes important enough to get fixed in a LTS distro.  

What are the disadvantages / "less/non performant" items associated with running Humble?

It seems Galactic G.5.3 does not publish /dock_status?

Context:  I have both Galactic and Humble Desktops configured, and have a Create3 arriving this week, presumably with Galactic firmware.  I need to decide if I start with the “stock” firmware, or immediately upgrade it to Humble.

(I do understand running an EOL code base.  My six year old GoPiGo3 robot Carl has been running 24/7 without issue for three years beyond its “Raspbian For Robots OS” EOL.  If it works don’t mess with it!)

**Comment by brianabouchard** [comment, marked as answer]
Hello! I can say that I have been using the Create 3 Robot with ROS 2 Humble relatively exclusively for the last 7-8 months. The only challenge I'm aware of is that when using Fast-DDS, the startup time is delayed by about 30 seconds (see release notes). I do believe the team is working on a fix to resolve this issue.

**Comment by shamlian** [comment]
I'm going to remove the note on https://iroboteducation.github.io/create3_docs/releases/overview/ since we fixed the performance issue in H.2.2 [(see release notes)](https://iroboteducation.github.io/create3_docs/releases/h_2_2/) in a PR, soon. Thanks for the issue. To @brianabouchard 's point, we are working to root-cause and solve the startup delay that was mentioned; because of this delay, I plan to leave `latest` pointing to the `galactic-latest` release, for now. That said, if a user wants to use Humble and doesn't mind the initial extra startup time, I don't know of a reason they shouldn't use it.

**Comment by shamlian** [comment]
Also, a note -- between Galactic and Humble, dock-related topics and actions changed names. Please see [the docs](https://iroboteducation.github.io/create3_docs/api/docking/) for details.
